### PR TITLE
Drawer Component: Pass Component To the Drawer Header

### PR DESCRIPTION
**Pull request by vintagepenguin**
Issue: https://github.com/mxenabled/mx-react-components/issues/495

## Shout out to @paniclater for a dope pairing session with this. 

# `GIF:` after most recent commit `(1/20/17)`
[https://gyazo.com/4b37d42babd16b15efefa5d190059fc4](https://gyazo.com/4b37d42babd16b15efefa5d190059fc4)

### CODE REVIEW

File: src/components/Drawer.js [hunk 1]
```diff
@@ -24,16 +24,12 @@ const Drawer = React.createClass({
     ]),
     duration: React.PropTypes.number,
     easing: React.PropTypes.array,
+    headerMenu: React.PropTypes.func,
```
Comment (phantomxc) [inline src/components/Drawer.js]: This can be `React.PropTypes.element`

**Comment by paniclater** [comment]
This is great! However, you need to add a check so if the prop isn't passed in the old behavior is preserved so it can be a non-breaking change

File: src/components/Drawer.js (hunk 1 above)
Comment (vintagepenguin) [inline src/components/Drawer.js]: ohhhh niccce pro-tip ⚡️

File: docs/components/DrawerDocs.js [hunk 2]
```diff
@@ -140,17 +150,72 @@ const DrawerDocs = React.createClass({
...
+      onClose={this._handleDrawerClose}
       title='Demo Drawer'
     >
-      //Content here
+      // Content Here
+
     </Drawer>
+
+    // Component To Pass As A Prop
+    function HeaderMenu ({ handleButtonClick, handleScrimClick, showSimpleSelectMenu = false }) {
```
Comment (paniclater) [inline docs/components/DrawerDocs.js]: is this defined 2 times?

Comment (paniclater) [inline docs/components/DrawerDocs.js]: I also think we should add the header menu as a separate component in `src/components/HeaderMenu.js`

File: src/components/Drawer.js [hunk 3]
```diff
@@ -161,7 +162,9 @@ const Drawer = React.createClass({
               <span style={styles.title}>
                 {this.props.title}
               </span>
-              {this._renderNav()}
+              <div style={styles.headerMenu}>
+                {this.props.headerMenu ? this.props.headerMenu : this._renderNav}
```
Comment (paniclater) [inline src/components/Drawer.js]: 👍

Comment (phantomxc) [inline src/components/Drawer.js]: this._renderNav() ?

File: docs/components/DrawerDocs.js (hunk 2 above)
Comment (vintagepenguin) [inline docs/components/DrawerDocs.js]: it's defined once in the code and once in the docs

File: src/components/Drawer.js (hunk 3 above)
Comment (vintagepenguin) [inline src/components/Drawer.js]: 💯  nicce catch

**Comment by paniclater** [comment]
:shipit:

**Comment by paniclater** [comment]
:shipit:

**Pull request merged**